tools/ml-audio-content-moderation/app/angular/src/app/histogram/histogram.component.ts

```diff
@@ -0,0 +1,155 @@
+import {
+  AfterViewInit,
+  Component,
+  ElementRef,
+  Input,
+  ViewChild,
+  ViewEncapsulation,
+} from '@angular/core';
+import * as d3 from 'd3';
+
+@Component({
+  selector: 'app-histogram',
+  templateUrl: './histogram.component.html',
+  styleUrls: ['./histogram.component.css'],
+  encapsulation: ViewEncapsulation.None,
+})
+/** Class to create histogram using D3 library. */
+export class HistogramComponent implements AfterViewInit {
+  @ViewChild('chart') private chartContainer: ElementRef;
+  @Input() private histogramData: number[] = [];
+  private readonly margin: any = { top: 20, bottom: 20, left: 20, right: 20 };
+  private chart: any;
+  private svg: any;
+  private width: number = 0;
+  private height: number = 0;
+  private data: number[];
+
+  constructor() {}
+
+  private createChart() {
+    let element = this.chartContainer.nativeElement;
+    this.width = element.offsetWidth - this.margin.left - this.margin.right;
+    this.height = 220 - this.margin.top - this.margin.bottom;
+    this.svg = d3
+      .select(element)
+      .append('svg')
+      .attr('width', this.width + this.margin.left + this.margin.right)
+      .attr('height', this.height + this.margin.top + this.margin.bottom);
+
+    this.chart = this.svg
+      .append('g')
+      .attr(
+        'transform',
+        'translate(' + this.margin.left + ',' + this.margin.top + ')'
+      );
+  }
+
+  private updateChart() {
```

Comment:
most of the variables inside this method don't look to be reassigned, in which case they can be const